src/command_modules/azure-cli-network/azure/cli/command_modules/network/_validators.py

```diff
@@ -232,6 +232,26 @@ def validate_dns_record_type(namespace):
             return
 
 
+def validate_er_peer_circuit(cmd, namespace):
+    from msrestazure.tools import resource_id, is_valid_resource_id
+
+    if not is_valid_resource_id(namespace.peer_circuit):
+        peer_id = resource_id(
+            subscription=get_subscription_id(cmd.cli_ctx),
+            resource_group=namespace.resource_group_name,
+            namespace='Microsoft.Network',
+            type='expressRouteCircuits',
+            name=namespace.peer_circuit,
+            child_type_1='peerings',
+            child_name_1=namespace.peering_name)
+
+    # if the circuit ID is provided, we need to append /peerings/{peering_name}
+    if namespace.peering_name not in peer_id:
```

Comment:
can we force customer to provide resource Id of private peering of peer circuit instead of handling it internally to keep it consistent with PowerShell

Reply:
Consistency with Powershell is not something we strive for.  Besides, the most likely case is that the user would just provide the name anyways rather than the ID.